EIPS/eip-8011.md

```diff
@@ -0,0 +1,172 @@
+
+```python
+def execute_transaction(self, transaction: NormalizedTransaction, effective_gas_price: int) -> int, narray: pass
+```
+
+### Block header extension
+
+The current header encoding is extended with a new 64-bit unsigned integer field named `max_gas_metered`. This integer corresponds to the total gas units used by the bottleneck resource, i.e., the resource with the largest used gas in the block. This variable is computed as follows:
+
+```python
+def compute_block_max_gas_metered(block: Block) -> int:
+    transactions = self.transactions(block)
+    block_gas_used_vector = array(0, 0, 0, 0, 0, 0)
+    for transaction in transactions:
+        gas_used, gas_used_vector = self.execute_transaction(transaction, effective_gas_price)
+        block_gas_used_vector += gas_used_vector
+    max_gas_metered = max(block_gas_used_vector)
+    return max_gas_metered
+```
+
+The header sequence with the new field is `[..., parent_beacon_block_root, requests_hash, max_gas_metered]`.
+
+### Block validity condition
+
+The block validity conditions are modified to replace `block.gas_used` with `block.max_gas_metered`:
+
+```python
+assert block.max_gas_metered <= block.gas_limit, 'invalid block: too much gas used'
+```
+
+### Base fee update rule
+
+The block validity conditions are modified to replace `parent.gas_used` with `parent.max_gas_metered`:
+
+```python
+gas_used_delta = parent.max_gas_metered - parent.gas_target
+```
+
+This change is compatible with the various transaction variants, such as [EIP-1559](./eip-1559.md), [EIP-4844](./eip-4844.md),  or [EIP-7999](./eip-7999.md).
+
+## Rationale
+
+### What are the benefits of this proposal?
+
+There are four main benefits of the proposal:
+
+1. Throughput gains: By decoupling resource limits, blocks can carry transactions that stress distinct resources simultaneously, improving packing efficiency.
+2. Finer-grained control over excessive resource usage: metering gas costs independently for each resource allows us to tailor each operation's gas cost to the actual resource limits.
+3. Keeps UX identical: users still specify one `gas_limit` and see one `gas_used` value.
+4. Simplicity: This simpler metering change lays the groundwork for eventual multidimensional pricing without disrupting the current fee market. Thus, this proposal serves as a stepping stone for future multidimensional gas pricing designs, such as [EIP-7999](./eip-7999.md).
+
+### Why are we choosing this resource split?
+
+Ethereum’s slot-based structure introduces a strict temporal constraint: all attestations must be processed, aggregated, and propagated within a single 12-second slot. This fact makes time a fundamental resource. To maintain network health, validators must execute blocks, validate them, and gossip attestations quickly enough to avoid missed slots and penalties.
```

Comment:
avoid mentioning al
```suggestion
Ethereum’s slot-based structure introduces a strict temporal constraint: all attestations must be processed, aggregated, and propagated within a single slot. This fact makes time a fundamental resource. To maintain network health, validators must execute blocks, validate them, and gossip attestations quickly enough to avoid missed slots and penalties.
```
avoid mentioning actual slot time because slot time might drastically reduce